### DISCUSSION TITLE
connection through udp stuck

### DISCUSSION THREAD

**Discussion by w2016561536** [bitcraze/discussions#193] [Q&A] [answered]
The console output below here：
`INFO:cfclient.gui:Disabling STL printouts  

INFO:cfclient.utils.input.inputreaders:Input readers: ['linuxjsdev', 'pysdl2']  

INFO:cfclient.utils.input.inputreaders:Successfully initialized [linuxjsdev]  

INFO:cfclient.utils.input.inputreaders:Could not initialize [pysdl2]: No SDL2 support on Linux  

INFO:cfclient.utils.input.inputinterfaces:Found interfaces: ['leapmotion', 'wiimote', 'zmqpull']  

INFO:cfclient.utils.input.inputinterfaces:Could not initialize [leapmotion]: Leap Motion library probably not installed (No module named 'leapsdk')  

INFO:cfclient.utils.input.inputinterfaces:Could not initialize [wiimote]: Missing cwiid (wiimote) driver No module named 'cwiid'  

INFO:cfclient.utils.config:Dist config read from /home/wyt/cf-client/crazyflie-clients-python/src/cfclient/configs/config.json  

INFO:cfclient.utils.config:Config file read from [/home/wyt/.config/cfclient/config.json]  

INFO:cfclient.utils.input.inputinterfaces:Could not initialize [zmqpull]: ZMQ input disabled in config file  

QSocketNotifier: Can only be used with threads started with QThread  

INFO:cfclient.utils.zmq_param:Biding ZMQ for parameters at tcp://*:1213  

INFO:cfclient.utils.zmq_led_driver:Biding ZMQ for LED driverat tcp://*:1214  

INFO:cfclient.utils.input:Using device blacklist [(VirtualBox|VMware)]  

INFO:cfclient.ui.main:Auto reconnect enabled: True  

INFO:cfclient.ui.tabs.QualisysTab:Switching Flight Mode to: FlightModeStates.DISCONNECTED  

INFO:cflib.crazyflie:Callback->Connection initialized[udp://192.168.43.42]  

INFO:cflib.crazyflie:We are connected[udp://192.168.43.42], request connection setup  

INFO:cflib.crazyflie:Callback->Connected to [udp://192.168.43.42]  

INFO:cflib.crazyflie.platformservice:Procotol version: 4  

INFO:cflib.crazyflie.toc:TOC for port [5] found in cache  

INFO:cflib.crazyflie:Log TOC finished updating`  

And the ui stuck at Connecting to "udp:xxxxxxx"

I have tried to run it in Ubuntu 22.04 LTS(Virtual Machine) , vm provided by bitcraze and windows 10 (physical machine) , but all of them doesn't work. All of them print the same debug info.
How do i solve it?
Thanks!

**Comment by w2016561536** [comment, marked as answer]
I have found the problem，after I rollback the rom，it solved

**Reply by knmcguire** [reply]
Hi! great you solved it, but we are still unclear on what you mean by 'rollback the rom'

**Reply by w2016561536** [reply]
The rom (or firmware) of the plane is modified by espressif to run the crazyflie on esp32. However espressif has update the core to 2021_01. And I have tried a older version of firmwork .It works.

**Reply by w2016561536** [reply]
Sorry to disturb u. I considered that the problem is the udpdriver of cfclient. However it turns out that the problem is created by espressif

**Reply by knmcguire** [reply]
ah that clarifies it a lot. Thanks!
